.github/workflows/release_deb.yml

```diff
@@ -0,0 +1,112 @@
+name: "Release DEB package"
+
+on:
+  release:
+    types:
+      - published
```

Comment:
It should be in the Release workflow, instead of a new workflow. And the most important, it should be triggered on tags instead of when a release is published.